agent/client.go

```diff
@@ -116,6 +117,23 @@ func (c *CanaryClient) CreateServer(srv *Server) (string, error) {
 	return respServer.UUID, nil
 }
 
+func (client *CanaryClient) FetchUpgradeablePackages() (map[string]string, error) {
+	respBody, err := client.get(ApiServerPath(client.server.UUID))
```

Comment:
I like the idea of just having an endpoint, tho if you make a convincing argument for parsing the existing one…